core/Plugin.php

```diff
@@ -451,6 +451,19 @@ public static function getPluginNameFromNamespace($namespaceOrClassName)
     }
 
     /**
+     * Override this method in your plugin class if you want your plugin to be loaded during tracking.
+     *
+     * Note: If you define your own dimension or handle a tracker event, your plugin will automatically
+     * be detected as a tracker plugin.
+     *
+     * @return bool
+     */
+    public function isTrackerPlugin()
```

Comment:
Should we mark this as `@internal` as it is not used by an external plugin currently (would be just easier to remove it then at some point as we will refactor those things at some point for sure)? Or maybe we could merge it to `plugin.json` as we have already `theme:true/false` there? Just a random thought...